internal/test/components/components.yaml

```diff
@@ -5,6 +5,38 @@ info:
   license:
     name: MIT
 paths:
+  /ensure-everything-is-referenced:
+    get:
+      operationId: ensureEverythingIsReferenced
+      description: |
+        This endpoint exists so that components can be created in this
+        spec and not be pruned
+    # TODO: figure out why uncommenting this causes failures
```

Comment:
Uncommenting this causes the following error

```sh
error generating code: error formatting Go code: 990:17: expected selector or type assertion, found '!=' (and 10 more errors)
```

I'm unsure exactly what it is but it seems like it might be an pre-existing issue with the code generation.

Reply:
I made some progress with this, the first problem was that the `ParmeterObject` was missing a `name`.

After fixing that, the new problem is that there are instances in both the server and client generated code where the generated code assumes that the parameter will be of type `string` but it is actually the `ParameterObject` struct.